[MRG] Fix in plot_out_of_core_classification.py

devashishd12 opened the pull request:
Just added conditions to check which one to use from `color_cycle` or `prop_cycle` depending on matplotlib version. Fix for #6016

devashishd12 committed:
Added a conditional statement to use prop_cycle or color_cycle depending on the matplotlib version.

lesteve commented on examples/applications/plot_out_of_core_classification.py:
```diff
@@ -398,7 +404,10 @@ def autolabel(rectangles):
 cls_names.append('Read/Parse\n+Feat.Extr.')
 cls_runtime.append(vectorizing_time)
 cls_names.append('Hashing\n+Vect.')
-bar_colors = rcParams['axes.color_cycle'][:len(cls_names)]
```
You can actually remove this line completely since bar_colors is defined earlier.

lesteve commented:
Maybe a more pragmatic approach is to do:

```python
bar_colors = ['b', 'g', 'r', 'c', 'm', 'y']
```

this way you don't have to check the matplotlib version at all.

Also use "[MRG]" at the beginning of your title to indicate you feel your PR is ready for merge.

dsquareindia commented:
Hey @lesteve thanks a lot for the input! Just had a few doubts. I think we can't use the latter approach to make `bar_colors` a list because the former approach returns a Cycler (not a list). Plus we need to dynamically cycle through a certain number of colors depending on the length and not all of them always. What do you think?

devashishd12 committed:
Small change

Will have to import full matplotlib to check version. Might lead to performance issues. Any other way to fix this?

srinivasreddy commented on examples/applications/plot_out_of_core_classification.py:
```diff
@@ -362,7 +366,10 @@ def plot_accuracy(x, y, x_legend):
 cls_runtime.append(total_vect_time)
 cls_names.append('Vectorization')
-bar_colors = rcParams['axes.color_cycle'][:len(cls_names)]
+if ver<'1.5.1':
```
though it works until single digit major version, it does lexicographic comparison. It fails for double digit numbers

devashishd12 committed:
Eliminated the use of color_cycle

Just made bar_colors a list.

dsquareindia commented on examples/applications/plot_out_of_core_classification.py:
```diff
@@ -362,7 +363,7 @@ def plot_accuracy(x, y, x_legend):
 cls_runtime.append(total_vect_time)
 cls_names.append('Vectorization')
-bar_colors = rcParams['axes.color_cycle'][:len(cls_names)]
+bar_colors = ['b','r','g','c','m','y']
```
Made bar_colors a list. Removed matplotlib version dependency.

dsquareindia commented:
@lesteve could you please check this once? Thanks!

lesteve commented:
> @lesteve could you please check this once? Thanks!

LGTM.

CircleCI failure is because of examples/ensemble/plot_partial_dependence.py and is unrelated (related to #5953). You can probably rebase on master if you want to get rid of it.

rvraghav93 commented:
Could you post the before/after plots and see if there are any changes. Once done :+1:  from me too!

dsquareindia commented:
| With `color_cycle` | Without `color_cycle` |
| --------------- | -------------------- |
| ![figure_1](https://cloud.githubusercontent.com/assets/9411999/11848140/eacc434c-a446-11e5-80bf-8cea69845eb5.png) | ![figure_1](https://cloud.githubusercontent.com/assets/9411999/11848152/fab2687c-a446-11e5-9d23-a0a7cac28c21.png) |
| ![figure_1-1](https://cloud.githubusercontent.com/assets/9411999/11848178/197fbfa2-a447-11e5-8a20-caa1bf5a0fbb.png) | ![figure_1-1](https://cloud.githubusercontent.com/assets/9411999/11848194/249f2b52-a447-11e5-93d1-8c00587393e3.png) |
| ![figure_2](https://cloud.githubusercontent.com/assets/9411999/11848207/3c792a20-a447-11e5-88f3-5a6be39dfaff.png) | ![figure_2](https://cloud.githubusercontent.com/assets/9411999/11848219/4734832e-a447-11e5-9490-bfd22bd8713e.png) |
| ![figure_3](https://cloud.githubusercontent.com/assets/9411999/11848227/55641766-a447-11e5-8e31-59c8afc9df9d.png) | ![figure_3](https://cloud.githubusercontent.com/assets/9411999/11848236/5eb41e88-a447-11e5-80f5-99a335765621.png) |

rvraghav93 commented:
The last plot on the right seems to be different. Could you check and correct it?

rvraghav93 commented:
I.e. the colours of pa and perceptron are interchanged...

devashishd12 committed:
Corrects colors for pa and perceptron

dsquareindia commented:
@rvraghav93 
### After the small fix:
![figure_1](https://cloud.githubusercontent.com/assets/9411999/11860634/caf6ab24-a49c-11e5-9eb5-fad0dfea892c.png)
![figure_1-1](https://cloud.githubusercontent.com/assets/9411999/11860686/5ae6f478-a49d-11e5-800c-e06aea7c4551.png)
![figure_2](https://cloud.githubusercontent.com/assets/9411999/11860688/60231f52-a49d-11e5-8d49-21531ba22b60.png)
![figure_3](https://cloud.githubusercontent.com/assets/9411999/11860689/657b44ac-a49d-11e5-91b9-47e540ce8ccf.png)

rvraghav93 commented:
Thanks! :+1:

dsquareindia commented:
@amueller could you please check this once? Thanks!

ogrisel commented:
LGTM. Fix the pep8  style issues, rebased, squashed and pushed to master.

ogrisel commented:
Here is the resulting commit: 80c770ffa03fe1a6980b02608bfad0aa0377a12a

dsquareindia commented:
@ogrisel thanks!
